pkg/common/cluster_actions.go

```diff
@@ -650,16 +640,6 @@ func (i *ClusterActionRunner) configureRealmClientScopes(obj *v1alpha1.KeycloakR
 		}
 	}
 
-	// If additions were made, rebuild the actual realm list.
-	if len(clientScopesToAdd) > 0 {
-		actionLogger.Info("Client scopes were added, rebuild client scope list.")
-		actualRealmClientScopes, err = i.keycloakClient.ListRealmClientScopes(realmName)
-		if err != nil {
-			return err
-		}
-		_, actualRealmClientScopeMap = prepareActualRealmClientScopes(actualRealmClientScopes)
-	}
-
 	// Do comparisons
 	for _, name := range clientScopesToCompare {
 		err = i.compareAndUpdateRealmClientScope(realmName, desiredRealmClientScopeMap[name], actualRealmClientScopeMap[name], actualRealmClientScopeMap, actionLogger)
```

Comment:
You should remove the `actualRealmClientScopeMap` argument from this function and the one for required actions, as they are not used in the function implementations (same reason you don't need to rebuild the mapping after adding a  new one).